lib/features/user/data/data_sources/data_access_objects/user_dao.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:hotfoot/features/user/data/models/user_model.dart';
+import 'package:meta/meta.dart';
+import 'package:sembast/sembast.dart';
+
+abstract class IUserDao {
+  Future<void> insert({@required UserModel userModel});
+  Future<void> update({@required UserModel userModel});
+  Future<void> delete({@required String id});
+  Future<List<String>> getPastOrderIds();
+  Future<List<String>> getPastOrderAddresses();
+  Future<void> addOrderId({@required String orderId});
+  Future<void> addAddress({@required String address});
+  Future<int> deleteAll();
+  Future<UserModel> getUserInformation();
+  Future<void> insertOrUpdate({UserModel userModel});
+}
+
+class UserDao implements IUserDao {
+  final Database database;
+  static const String _USER_STORE_NAME = 'users';
+  final _userStore = stringMapStoreFactory.store(_USER_STORE_NAME);
+
+  UserDao({@required this.database});
+
+  @override
+  Future<void> insert({@required UserModel userModel}) async {
+    return await _userStore
+      .record(userModel.id)
+      .put(database, userModel.toJson());
+  }
+
+  @override
+  Future<void> update({@required UserModel userModel}) async {
+    final finder = Finder(filter: Filter.byKey(userModel.id));
+    return await _userStore.update(
+      database,
+      userModel.toJson(),
+      finder: finder,
+    );
+  }
+
+  @override
+  Future<void> delete({@required String id}) async {
+    return await _userStore.record(id).delete(database);
+  }
+
+  // If you have a more optimal way of doing this please update this code

```

Comment:
If we make the list a sub-collection it can be done.
But I think it's overkill for this. This looks good enough.